src/components/choose-locale/change-locale.component.tsx

```diff
@@ -0,0 +1,49 @@
+import { refetchCurrentUser } from "@openmrs/esm-api";
+import React, { useEffect, useState } from "react";
+import { updateUserProperties } from "./change-locale.resource";
+
+interface ChangeLocaleProps {
+  allowedLocales: Array<string>;
+  user: any;
+}
+
+export const ChangeLocale: React.FC<ChangeLocaleProps> = ({
+  allowedLocales,
+  user
+}) => {
+  const [userProps, setUserProps] = useState(user.userProperties);
+  const options = allowedLocales?.map(locale => (
+    <option value={locale} key={locale}>
+      {locale}
+    </option>
+  ));
+
+  useEffect(() => {
+    if (user.userProperties.defaultLocale !== userProps.defaultLocale) {
+      updateUserProperties(user.uuid, userProps, new AbortController()).then(
+        response => {
+          if (response.ok) {
+            refetchCurrentUser();
+          }
+        }
+      );
```

Comment:
Return a cancelling function using the AbortController. See https://github.com/openmrs/openmrs-esm-login/blob/ba86336f68ddc4fa364534b605a4184fea5ac1eb/src/choose-location/choose-location.component.tsx#L71 for example.

Reply:
@samuelmale This is the only change I'm going to say is *strictly required* before I approve this.

Reply:
Nice catch @brandones 

There are many places we need to fix this. Most especially in the patient registration esm.